blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxMetadataIO.java

```diff
@@ -0,0 +1,162 @@
+
+        Session session = DatastaxIO.getSession();
+
+        try {
+
+            Select.Where select = select()
+                    .all()
+                    .from( CassandraModel.CF_METRICS_METADATA_NAME )
+                    .where( eq( KEY, locator.toString() ));
+
+            List<Row> results = session.execute( select ).all();
+
+            Map<String, String> values = new HashMap<String, String>();
+
+            for ( Row row : results ) {
+                if ( LOG.isDebugEnabled() ) {
+                    LOG.debug( "Read metrics_metadata: " +
+                            row.getString( KEY ) +
+                            row.getString( COLUMN1 ) +
+                            serDes.deserialize(  row.getBytes( VALUE ) ) );
+                }
+
+                values.put( row.getString( COLUMN1 ), serDes.deserialize(  row.getBytes( VALUE ) ) );
+            }
+
+            return values;
+        }
+        finally {
+            ctx.stop();
+        }
+    }
+
+    @Override
+    public Table<Locator, String, String> getAllValues( Set<Locator> locators ) throws IOException {
+
+        Timer.Context ctx = Instrumentation.getReadTimerContext( CassandraModel.CF_METRICS_METADATA_NAME );
+
+        Session session = DatastaxIO.getSession();
+
+        try {
+
+            List<String> query = new ArrayList<String>();
+
+            for( Locator l : locators) {
+                query.add( l.toString() );
+            }
+
+            Select.Where select = select()
+                    .all()
+                    .from( CassandraModel.CF_METRICS_METADATA_NAME )
+                    .where( in( KEY, query ) );
+
+            List<Row> results = session.execute( select ).all();
+
+            Table<Locator, String, String> metaTable = HashBasedTable.create();
+
+            for ( Row row : results ) {
+                if ( LOG.isDebugEnabled() ) {
+                    LOG.debug( "Read metrics_metadata: " +
+                            row.getString( KEY ) +
+                            row.getString( COLUMN1 ) +
+                            serDes.deserialize(  row.getBytes( VALUE ) ) );
+                }
+
+                metaTable.put( Locator.createLocatorFromDbKey( row.getString( KEY ) ), row.getString( COLUMN1 ), serDes.deserialize(  row.getBytes( VALUE ) ) );
+            }
+
+            return metaTable;
+        }
+        finally {
+            ctx.stop();
+        }
+    }
+
+    @Override
+    public void put( Locator locator, String key, String value ) throws IOException {
+
+        Timer.Context ctx = Instrumentation.getWriteTimerContext( CassandraModel.CF_METRICS_METADATA_NAME );
+
+        Session session = DatastaxIO.getSession();
+
+        try {
+            Insert insert = insertInto( CassandraModel.CF_METRICS_METADATA_NAME )
+                    .value( KEY, locator.toString() )
+                    .value( COLUMN1, key )
+                    .value( VALUE, serDes.serialize( value ) );
```

Comment:
Datastax recommends the preparedstatement if you are executing it more than once.  I'm for updating our stuff to preparedstatement if we dont' think it will add too much work.  Otherwise we can do it in a later story.